src/Products/Viewer/Controllers/ViewerApiController.cs

```diff
@@ -430,149 +288,45 @@ private LoadDocumentEntity LoadDocument(PostedDataEntity postedData, bool loadAl
+                _pages.Add(pageStream);
 
-        private List<string> GetAllPagesContent(string password, string documentGuid)
-        {
-            List<string> allPages = new List<string>();
-            if (globalConfiguration.Viewer.GetIsHtmlMode())
-            {
-                HtmlOptions htmlOptions = new HtmlOptions();
-                SetOptions(htmlOptions, password, 0);
-                // get page HTML              
-                var pages = this.GetHandler().GetPages(documentGuid, htmlOptions);
-                for (int i = 0; i < pages.Count; i++)
-                {
-                    allPages.Add(pages[i].HtmlContent);
-                }
-            }
-            else
-            {
-                ImageOptions imageOptions = new ImageOptions();
-                SetOptions(imageOptions, password, 0);
-                var pages = this.GetHandler().GetPages(documentGuid, imageOptions);
-                for (int i = 0; i < pages.Count; i++)
-                {
-                    byte[] bytes;
-                    using (var memoryStream = new MemoryStream())
-                    {
-                        pages[i].Stream.CopyTo(memoryStream);
-                        bytes = memoryStream.ToArray();
-                    }
-                    string encodedImage = Convert.ToBase64String(bytes);
-                    allPages.Add(encodedImage);
-                }
+                return pageStream;
             }
-            return allPages;
-        }
 
-        private void SetOptions(HtmlOptions options, string password, int pageNumber)
-        {
-            Watermark watermark = null;
-            if (!String.IsNullOrEmpty(globalConfiguration.Viewer.GetWatermarkText()))
+            public void ReleasePageStream(int pageNumber, Stream pageStream)
             {
-                // Set watermark properties
-                watermark = new Watermark(globalConfiguration.Viewer.GetWatermarkText());
-                watermark.Color = System.Drawing.Color.Blue;
-                watermark.Position = WatermarkPosition.Diagonal;
-                watermark.Width = 100;
+                //Do not release page stream as we'll need to keep the stream open
             }
-            options.EmbedResources = true;
-            // set password for protected document
-            if (!string.IsNullOrEmpty(password))
-            {
-                options.Password = password;
-            }
-            if (watermark != null)
-            {
-                options.Watermark = watermark;
-            }
-            if (pageNumber != 0)
-            {
-                options.PageNumber = pageNumber;
-                options.CountPagesToRender = 1;
-            }
-            options.CellsOptions.ShowGridLines = globalConfiguration.Viewer.GetShowGridLines();
         }
 
-        private void SetOptions(ImageOptions options, string password, int pageNumber)
+        private PageDescriptionEntity GetPageDescriptionEntities(Page page)
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Make 'GetPageDescriptionEntities' a static method.](https://app.codacy.com/gh/groupdocs-viewer/GroupDocs.Viewer-for-.NET-MVC/pullRequest?prid=5242614)